scripts/missions/toau/40_Unraveling_Reason.lua

```diff
@@ -0,0 +1,108 @@
+-----------------------------------
+-- Unraveling Reason
+-- Aht Uhrgan Mission 40
+-----------------------------------
+-- !addmission 4 39
+-- Pherimociel : !pos -31.627 1.002 67.956 243
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/missions')
+require('scripts/globals/npc_util')
+require('scripts/globals/keyitems')
+require('scripts/globals/interaction/mission')
+require('scripts/globals/zone')
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.TOAU, xi.mission.id.toau.UNRAVELING_REASON)
+
+mission.reward =
+{
+    title       = xi.title.ENDYMION_PARATROOPER,
+    nextMission = { xi.mission.log_id.TOAU, xi.mission.id.toau.LIGHT_OF_JUDGMENT },
+}
+
+mission.sections =
+{
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId
+        end,
+
+        [xi.zone.RULUDE_GARDENS] =
+        {
+            ['Pherimociel'] =
+            {
+                onTrigger = function(player, npc)
+                    local questTimer = mission:getVar(player, 'Timer')
+
+                    if questTimer == 0 then
+                        return mission:progressEvent(10099)
+                    elseif
+                        VanadielUniqueDay() >= questTimer and
+                        not mission:getMustZone(player)
+                    then
+                        return mission:progressEvent(10098)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10098] = function(player, csid, option, npc)
+                    if option == 1 then
+                        -- Title can be obtained immediately following this event.  Since these are stringed
+                        -- together, set it at the end for clarity in mission rewards.
+                        player:setPos(0, 0, 0, 0, 51)
```

Comment:
Wouldnt this pos get overwritten to default pos defined in Zone? if some, and becouse of certain PRs like this #1222 shouldnt we set the actual pos we want just on case?

Reply:
Its there on purpose to trigger a download, the onZoneIn return forces the correct series of events.

Reply:
yeah, i get that, I'm not arguing against it. but i meant the first 3 values that determine the x y z.

the 5th value is the zone id, witch when set forces a zone change no matter if the target zone is the same as current zone.

So what I meant is, should we be setting the correct x,y,z just so the onZoneIn invalid pos fix wont kick in if the target pos in said zone is ever changed?

Reply:
In that case, we really don't care where we go in the zone, because there's several back-to-back events that are triggered in sequence.  We're forcing a zone change, and then handling it on the back end (they're all on zone ins, so a disconnect would resume where it left off)